### DISCUSSION TITLE
st-flash or stm32flash support

### DISCUSSION THREAD

**Discussion by Evidlo** [stm32duino/Arduino_Core_STM32#1569] [General]
Just wanted to create an issue to track discussion about supporting the st-flash or stm32flash utilities.  These tools seem to support the devices on most of the popular board variants.

- [stm32flash supported boards](https://sourceforge.net/p/stm32flash/code/ci/master/tree/dev_table.c#l88) (GPL2)
- [st-flash supported boards](https://github.com/stlink-org/stlink/blob/develop/doc/devices_boards.md) (BSD3)

There was some discussion [here](https://github.com/stm32duino/Arduino_Core_STM32/issues/675) about bringing back the old tools.  Why does it matter if the tools support all devices?  If a particular device isn't supported, surely it won't show up in the "Upload method" menu? This would also open up the possibility of shipping all necessary tools with the core due to the licensing and small size of the tools.

I run a microcontrollers course based around AVRs (and looking to expand), and the dependency on STM32CubeProgrammer is a showstopper for us.

**Comment by fpistm** [comment]
Hi,

> it won't show up in the "Upload method" menu?

Unfortunately not as the way to declare upload method in Arduino is very limited and so this will bring lot of support because user would like to use it while it is not supported. Split then per support possibilities is also not a good solution as it would request more complexity in the menu and to update it regularly.

I've open a ST internal issue to be able to deliver it but currently I have no date.

If you really want to use it it is possibles to customize the core to support it as stated in the link you provide.
